Sming/Arch/Esp8266/Components/custom_heap/component.mk

```diff
@@ -7,17 +7,14 @@ COMPONENT_INCDIRS		:= umm_malloc/src umm_malloc/includes/c-helper-macros
 
 COMPONENT_DEPENDS		:= esp8266
 
-LIBMAIN					= mainmm
-LIBMAIN_SRC				= $(COMPONENT_LIBDIR)/libmainmm.a
-COMPONENT_TARGETS		+= $(LIBMAIN_SRC)
-
 COMPONENT_VARS			+= UMM_POISON_CHECK
 ifeq ($(UMM_POISON_CHECK),1)
 GLOBAL_CFLAGS			+= -DUMM_POISON_CHECK
 endif
 
-# Make copy of libmain and remove mem_manager.o module
-$(COMPONENT_RULE)$(LIBMAIN_SRC): $(SDK_LIBDIR)/libmain.a
-	$(info Enabling custom heap implementation)
-	$(Q) cp $^ $@
-	$(Q) $(AR) -d $@ mem_manager.o
+# remove mem_manager.o module from libmain of SDK
+define LIBMAIN_COMMANDS +=
```

Comment:
@aemseemann Please, take out of this PR all changes that are not related to the main goal of this PR. If the removed changes are still needed for this PR then add them as new PRs and we will review and merge them before this one.

Reply:
-> See #1959

The reason for this modifications is that the logic to prepare libmain in rboot/component.mk  became a bit unmanageable, so I felt it would be a good idea to make the whole libmain generation a bit more flexible